backend/internal/auth/sessionmanager.go

```diff
@@ -28,12 +44,20 @@ type Session struct {
 type SessionManager struct {
 	store   map[SessionID]Session
 	timeout time.Duration // absolute timeout for individual sessions
-	mtx     sync.RWMutex
+	mtx     sync.RWMutex  // mutex for store
+
+	// contextKey is the key used to set and retrieve session data from a context.Context
+	contextKey contextKey
 }
 
 // NewSessionManager creates a new *SessionManager
 func NewSessionManager(timeout time.Duration) *SessionManager {
-	return &SessionManager{store: make(map[SessionID]Session), timeout: timeout}
+	ck, err := newContextKey()
+	if err != nil {
+		// Fail here, this is an operating system error plus the app cannot function without contextKey
+		panic(err)
```

Comment:
Why panic instead of returning an error here?

Reply:
This is only run once during startup, and without it the auth protection middlewear won't function. So I figured best to just panic

Reply:
This section is being simplified so the `panic` will be removed, see https://github.com/ibeckermayer/teleport-interview/pull/7#discussion_r557701184